addons/point_of_sale/controllers/main.py

```diff
@@ -11,15 +11,11 @@
 class PosController(http.Controller):
 
     @http.route('/pos/web', type='http', auth='user')
-    def a(self, debug=False, **k):
-        cr, uid, context, session = request.cr, request.uid, request.context, request.session
-
+    def pos_web(self, debug=False, **k):
         # if user not logged in, log him in
-        PosSession = request.registry['pos.session']
-        pos_session_ids = PosSession.search(cr, uid, [('state','=','opened'),('user_id','=',session.uid)], context=context)
-        if not pos_session_ids:
-            return werkzeug.utils.redirect('/web#action=point_of_sale.action_pos_session_opening')
-        PosSession.login(cr, uid, pos_session_ids, context=context)
-        
+        pos_session = request.env['pos.session'].search([('state', '=', 'opened'), ('user_id', '=', request.session.uid)])
```

Comment:
Variable name should be plural,
You are not set attribute limit=1 so it may be get more then one record,
So.....
variable name should be ***pos_sessions*** instead of ***pos_session***
![plural](https://cloud.githubusercontent.com/assets/14940179/12124724/4ab95972-b3a4-11e5-81c8-1f6fb6f00256.jpg)
.